streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -16,71 +16,186 @@
+                                internalProcessorContext.currentNode().name(),
+                                buffer.numRecords(), maxRecords,
+                                buffer.bufferSize(), maxBytes
+                            ));
+                    }
+                }
+            }
+        }
     }
 
-    @Override
-    public String toString() {
-        return "KTableSuppressProcessor{" +
-            "suppress=" + suppress +
-            ", keySerde=" + keySerde +
-            ", valueSerde=" + valueSerde +
-            '}';
+    private KeyValue<TimeKey<K>, ContextualRecord<Change<V>>> drainExpiredRecords(
+        final long expiryTime,
+        final AutoCloseableIterator<KeyValue<TimeKey<K>, ContextualRecord<Change<V>>>> iterator) {
+
+        while (iterator.hasNext()) {
+            final KeyValue<TimeKey<K>, ContextualRecord<Change<V>>> next = iterator.next();
+            final long bufferTime = next.key.time();
+            if (bufferTime <= expiryTime) {
+                evictAndEmit(iterator, next);
+            } else {
+                return next;
+            }
+        }
+        return null;
     }
 
-    public static class NotImplementedException extends RuntimeException {
-        NotImplementedException() {
-            super();
+    private void evictAndEmit(final AutoCloseableIterator<KeyValue<TimeKey<K>, ContextualRecord<Change<V>>>> iterator,
+                              final KeyValue<TimeKey<K>, ContextualRecord<Change<V>>> next) {
+        final K key = next.key.key();
+        final ContextualRecord<Change<V>> contextualRecord = next.value;
+        // to be safe, forward before we delete
+        setNodeAndForward(key, contextualRecord);
+        iterator.remove();
+    }
+
+    private boolean overCapacity() {
+        return buffer.numRecords() > maxRecords || buffer.bufferSize() > maxBytes;
+    }
+
+    private void setNodeAndForward(final K key, final ContextualRecord<Change<V>> contextualRecord) {
+        if (shouldForward(contextualRecord.value())) {
+            final ProcessorRecordContext prevRecordContext = internalProcessorContext.recordContext();
+            internalProcessorContext.setRecordContext(contextualRecord.recordContext());
+            try {
+                internalProcessorContext.forward(key, contextualRecord.value());
+            } finally {
+                internalProcessorContext.setRecordContext(prevRecordContext);
+            }
+        }
+    }
+
+    private boolean shouldForward(final Change<V> value) {
+        return !(value.newValue == null && shouldSuppressTombstones);
+    }
+
+    private long computeRecordSize(final K key, final Change<V> value, final ProcessorRecordContext recordContext) {
+        if (maxBytes == Long.MAX_VALUE) {
+            // if the buffer isn't size-constrained, we don't need to compute the record size.
+            return 0L;
+        } else {
+            long size = 0L;
```

Comment:
we could use a test to hit this branch as well, but IMHO it's a lower priority than the others mentioned above.

Reply:
This line is actually gone now. But if it were still there, I'd agree with you.